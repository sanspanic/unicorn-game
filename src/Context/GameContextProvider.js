import React, { useState } from "react";
import GameContext from "./GameContext";

const GameContextProvider = ({ children }) => {
  const initialState = {
    status: "won",
    difficulty: 0,
    size: 15,
    name: "",
  };
  const [gameData, setGameData] = useState(initialState);

  //restarts game by setting gameData.status back to "setup" & redirecting to homepage
  const playAgain = () => {
    setGameData({ ...gameData, status: "setup" });
  };

  return (
    <GameContext.Provider value={{ gameData, setGameData, playAgain }}>
      {children}
    </GameContext.Provider>
  );
};

export default GameContextProvider;
