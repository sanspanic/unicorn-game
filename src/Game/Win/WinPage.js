import React, { useContext } from "react";
import GameContext from "../../Context/GameContext";
import { useHistory } from "react-router-dom";
import WinUnicornImage from "./WinUnicornImage";

const WinPage = () => {
  const { playAgain } = useContext(GameContext);
  const history = useHistory();

  const handlePlayAgain = () => {
    history.push("/");
    playAgain();
  };

  return (
    <div className="flex flex-col justify-evenly items-center p-10 h-screen">
      <div className="font-mono text-xl text-center">
        You reached the rainbow!
      </div>
      <WinUnicornImage />

      <button
        className="shadow-xl py-2 px-3 border-4 border-purple-700 rounded gradient-pink text-white font-mono"
        onClick={handlePlayAgain}
      >
        Restart
      </button>
    </div>
  );
};

export default WinPage;
