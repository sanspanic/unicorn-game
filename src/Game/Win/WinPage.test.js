import React from "react";
import { fireEvent, render } from "@testing-library/react";
import WinPage from "./WinPage";
import GameContextProvider from "../../Context/GameContextProvider";

test("renders without crashing", () => {
  render(
    <GameContextProvider>
      <WinPage />
    </GameContextProvider>
  );
});

test("matches snapshot", () => {
  const { asFragment } = render(
    <GameContextProvider>
      <WinPage />
    </GameContextProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});

//test fails :(
/* test("button click restarts game", () => {
  const { getByText } = render(
    <GameContextProvider>
      <WinPage />
    </GameContextProvider>
  );
  const button = getByText("Restart");
  //below line breaks test: no history object
  fireEvent.click(button);
}); */
