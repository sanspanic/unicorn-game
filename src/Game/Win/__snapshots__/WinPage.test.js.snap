// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-evenly items-center p-10 h-screen"
  >
    <div
      class="font-mono text-xl text-center"
    >
      You reached the rainbow!
    </div>
    <div
      class="react-floater-animated"
    >
      <img
        class="mx-auto sm:w-9/12 md:w-6/12 lg:w-4/12"
        src="WinGame.png"
      />
    </div>
    <button
      class="shadow-xl py-2 px-3 border-2 border-purple-700 rounded gradient-pink text-white font-mono"
    >
      Restart
    </button>
  </div>
</DocumentFragment>
`;
